### PR TITLE
Support device specifier in image captioning example (`predict.py`) 

### LINKED ISSUE #7205: ChainerX example support: image_captioning: predict.py

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7205

Part of #6661

~Depends on #7204~

### CODE REVIEW

File: examples/image_captioning/predict.py [hunk 1]
```diff
@@ -26,12 +28,27 @@ def main():
...
+    group = parser.add_argument_group('deprecated arguments')
+    group.add_argument('--gpu', '-g', dest='device',
+                       type=int, nargs='?', const=0,
+                       help='GPU ID (negative value indicates CPU)')
     args = parser.parse_args()
 
+    device = chainer.get_device(args.device)
+    if device.xp is chainerx:
+        sys.stderr.write('This example does not support ChainerX devices.\n')
+        sys.exit(1)
```
Comment (hvy) [inline examples/image_captioning/predict.py]: Is it the case that it actually works for ChainerX if `--rnn nsteplstm` ? I image that's why you also added `atleast_1d`. How about only existing for `--rnn lstm`?

Comment (niboshi) [reply examples/image_captioning/predict.py]: I don't think it's so productive to fine-tune the error to look for the cases in which ChainerX works.

Comment (hvy) [reply examples/image_captioning/predict.py]: I think that is fine too but we should in that case not introduce `_atleast_1d` as it's never called.

Comment (niboshi) [reply examples/image_captioning/predict.py]: That's true. Removed `_atleast_1d`.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**